tensorflow/core/util/gpu_kernel_helper.h

```diff
@@ -45,6 +45,23 @@ limitations under the License.
 #define GPU_AXIS_KERNEL_LOOP(i, n, axis) \
   for (int i : ::tensorflow::GpuGridRange##axis<int>(n))
 
+#if TENSORFLOW_USE_ROCM
+
+#define cub hipcub
```

Comment:
I’ll need to sit behind a computer to check this header again. Most of cuda to hipnchanges don’t sound correct to me

Reply:
Please elaborate. I am happy to add more advanced macros, however, these simple define changes do compile correctly.

Reply:
One of the goal is to try restrict exposure to `cuda` for less potential legal issue. I'd feel more comfortable putting `#if/#elif` blocks in the operators.